csharp-algorithms/FindAllPositionsOfAString/FindAllPositionsOfAString/Algorithms/SearchUsingBruteForceAlgorithm.cs

```diff
@@ -0,0 +1,46 @@
+﻿using FindAllPositionsOfAString.Algorithms.Interfaces;
+using System.Buffers;
+using System.Reflection.Metadata.Ecma335;
+
+namespace FindAllPositionsOfAString.Algorithms;
+
+public class SearchUsingBruteForceAlgorithm : SearchBase, ISearcher
+{
+    public List<int> FindAll(string text, string searchText)
+    {
+        List<int> positions = [];
+        var textSpan = text.AsSpan();
+        var searchTextSpan = searchText.AsSpan();
+
+        var textLength = textSpan.Length;
+        var searchLen = searchTextSpan.Length;
+
+        Func<char, char, bool> AreEqualCharacters = 
+            CaseSensitive ? AreEqualCharactersSensitive : AreEqualCharactersInsensitive;
+
+        int searchPosition = 0;
+        for (var textPosition = 0; textPosition <= textLength - searchLen; textPosition++)
+        {
+            for (searchPosition = 0; searchPosition < searchLen; searchPosition++)
+            {
+                if (!AreEqualCharacters(textSpan[textPosition + searchPosition], searchTextSpan[searchPosition]))
+                    break;
+            }
+
+            if (searchPosition == searchLen)
+            {
+                positions.Add(textPosition);
+                if (SkipWholeFoundText)
+                    textPosition += searchLen - 1;
+            }
+        }
+
+        return positions;
+    }
+
+    protected bool AreEqualCharactersSensitive(char a, char b) 
+        => a == b;
+
+    protected bool AreEqualCharactersInsensitive(char a, char b) 
+        => char.ToLowerInvariant(a) == char.ToLowerInvariant(b);
```

Comment:
Move to the base class since this is being repeated both here and in the KMP code.

Reply:
Corrected